lib/application.js

```diff
@@ -559,6 +560,43 @@ app.listen = function(){
   return server.listen.apply(server, arguments);
 };
 
+/*
+ * Set a tracer at the application level. Tracer will be activated on
+ * The tracer function must have the following signature:
+ *
+ *  function (app, req, res, event, date, args) ...
+ *
+ * @param {function} The tracer to set.
+ * @return {app} for chaining.
+ * @api public
+ */
+
+app.instrument = function(tracer){
+  if (tracer === undefined || typeof tracer  !== 'function')
+    throw new Error('instrument expects a function');
+  this.tracers.push(tracer);
+  return this;
+};
+
+/*
+ * Call all tracers set on the application. Add context information:
+ * running app, request, response and date.
+ *
+ * @param {Response} Response that fires the tracing event.
+ * @param {Event} The event to trace.
+ * @param {Array} Arguments to transmit to the tracker.
+ *
+ * @api private
+ */
+
+app._callTracers = function(res, event, args){
+  var date = new Date();
+  var app = this;
+  this.tracers.forEach(function(tracer){
```

Comment:
I suggest that we use for loop to avoid the performance penalty of closures. See http://jsperf.com/array-foreach-vs-for-loop.